eval-0trace/parse-each.go

```diff
@@ -0,0 +1,209 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"flag"
+	"github.com/go-ping/ping"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const zeroTraceBeginString = "TRACE RESULTS"
+
+var (
+	InfoLogger *log.Logger
+	EvalLogger *log.Logger
+)
+
+type Hops struct {
+	Hop int
+	IP  string
+}
+
+type Result struct {
+	Timestamp  string
+	ClientIP   string
+	Traceroute []Hops
+	Lasthop    string
+	Lastping   RtItem
+	ClientPing RtItem
+}
+
+type AvgRTTcompare struct {
+	Timestamp  string
+	ClientIP   string
+	LastHop    float64
+	ClientHop  float64
+	Difference float64
+}
+
+const ICMPCount = 5
+const ICMPTimeout = time.Second * 10
+
+type RtItem struct {
+	IP        string
+	PktSent   int
+	PktRecv   int
+	PktLoss   float64
+	MinRtt    float64
+	AvgRtt    float64
+	MaxRtt    float64
+	StdDevRtt float64
+}
+
+// IcmpPing conducts the ICMP ping measurement, code from proxy-detection-server
+func IcmpPing(ip string) (RtItem, error) {
+	pinger, err := ping.NewPinger(ip)
+	if err != nil {
+		log.Println(err)
+		return RtItem{}, err
+	}
+	pinger.Count = ICMPCount
+	pinger.Timeout = ICMPTimeout
+	err = pinger.Run() // Blocks until finished.
+	if err != nil {
+		log.Println(err)
+		return RtItem{}, err
+	}
+	stat := pinger.Statistics()
+	icmp := RtItem{ip, stat.PacketsSent, stat.PacketsRecv, stat.PacketLoss, fmtTimeMs(stat.MinRtt), fmtTimeMs(stat.AvgRtt), fmtTimeMs(stat.MaxRtt), fmtTimeMs(stat.StdDevRtt)}
+	return icmp, nil
+
+}
+
+// calcAvgDifference calculates the difference between the AvgRtt of the two given RtItem objects
+func calcAvgDifference(last RtItem, client RtItem) (float64, bool) {
```

Comment:
For extra clarity, we could also implement this as a method, i.e.:
```go
func (r1 *RtItem) avgDiff(r2 *RtItem) (floast64, bool) {
  ...
}
```

Also, it wasn't immediately clear to me why the function would return another variable.

Reply:
I've refactored this part of the code to include [this check](https://github.com/brave-experiments/proxy-detection-server/pull/6#discussion_r926118443) so I've called it `checkValidEvalResult` and return a diff (in avg RTT) and a bool now (bool is false if result is not valid for evaluation)